auth/src/ios/credential_ios.mm

```diff
@@ -197,11 +204,18 @@ @implementation PhoneListenerDataObjC
 
 // static
 bool GameCenterAuthProvider::IsPlayerAuthenticated() {
+  /**
+   Linking GameKit.framework without using it on macOS results in App Store rejection.
+   Thus we don't link GameKit.framework to our SDK directly. `optionalLocalPlayer` is used for
+   checking whether the APP that consuming our SDK has linked GameKit.framework. If not, 
+   early out.
+   **/
+  GKLocalPlayer *_Nullable optionalLocalPlayer = [[NSClassFromString(@"GKLocalPlayer") alloc] init];
   // If the GameKit Framework isn't linked - early out.
-  if ([GKLocalPlayer class] == nullptr) {
+  if (optionalLocalPlayer == nullptr) {
     return false;
   }
-  GKLocalPlayer *localPlayer = [GKLocalPlayer localPlayer];
+  __weak GKLocalPlayer *localPlayer = [[optionalLocalPlayer class] localPlayer];
```

Comment:
What you need to do here is basically
1. Get the class property [localPlayer](https://developer.apple.com/documentation/gamekit/gklocalplayer/1515401-localplayer?language=objc) from GKLocalPlayer, which is an instance of GKLocalPlayer
2. Get [isAuthenticated](https://developer.apple.com/documentation/gamekit/gklocalplayer/1515402-isauthenticated) from the instance of GKLocalPlayer

I think this can be replaced with [Objective-C Runtime](https://developer.apple.com/documentation/objectivec/objective-c_runtime). You may need to dig a bit deeper to figure out which functions to use.

I wonder why we have this public interface, while there is no such API for iOS SDK
https://firebase.google.com/docs/reference/ios/firebaseauth/api/reference/Classes/FIRGameCenterAuthProvider

I suggest you to contact iCore team if you want to get a quick answer for this.

Reply:
1. We do not need to remove this API now.
2. Seems like you are following the pattern in this https://github.com/firebase/firebase-ios-sdk/pull/2355. In that case, you change should be fine.